File: LeetCode Solutions/February LeetCoding Challenge/Arithmetic Slices.java
class Solution {
    public int numberOfArithmeticSlices(int[] A) {
        int arithmeticSlices = 0, combination = 0, lastDifference = 0;
        for(int i=0; i<A.length-1; i++){
            int difference = A[i+1] - A[i];
            if(i!=0 && difference == lastDifference){
                combination += 1;
                arithmeticSlices += combination;
            } else{
                combination = 0;
            }
            lastDifference = difference;
        }
        return arithmeticSlices;
    }
}

File: LeetCode Solutions/February 2020 LeetCoding Challenge/Arithmetic Slices.py
class Solution:
    def numberOfArithmeticSlices(self, A: List[int]) -> int:
        arithmeticSlices = combination = lastDifference = 0
        for i in range(len(A)-1):
            difference = A[i+1] - A[i]
            if(i!=0 and difference == lastDifference):
                combination += 1
                arithmeticSlices += combination
            else:
                combination = 0
            lastDifference = difference
        return arithmeticSlices